addons/hr_recruitment/models/hr_applicant.py

```diff
@@ -361,6 +361,11 @@ def write(self, vals):
             self._update_employee_from_applicant()
         return res
 
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_linked_employee(self):
+        if any(applicant.emp_id for applicant in self):
```

Comment:
Accessing `self.emp_id` will return a recordset of all employees from the applicants in self, doing exactly the same thing
```suggestion
        if self.emp_id:
```